### PR TITLE
[next][build-utils] Update to re-use build result type from build-utils

### PR DESCRIPTION
**Pull request by ijjk**
### Related Issues

We were previously using a custom type in Next.js for the build result although there is an existing type for this in the build-utils package which we can centralize around. 

x-ref: https://github.com/vercel/vercel/pull/7854#issuecomment-1133934764

#### Tests

- [ ] The code changed/added as part of this PR has been covered with tests
- [ ] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

**Review by AndyBitz — APPROVED** [review]

File: packages/build-utils/src/types.ts [hunk 1]
```diff
@@ -352,11 +352,41 @@ export interface BuilderV3 {
...
+};
+
 export interface Images {
   domains: string[];
+  remotePatterns?: RemotePattern[];
   sizes: number[];
   minimumCacheTTL?: number;
   formats?: ImageFormat[];
+  dangerouslyAllowSVG?: boolean;
+  contentSecurityPolicy?: string;
```
Comment (TooTallNate) [inline packages/build-utils/src/types.ts:389]: These should be added to the type in `@vercel/build-utils` instead, and this interface should be able to be removed as well.

Comment (ijjk) [reply packages/build-utils/src/types.ts:389]: Are the types listed twice in build-utils?

Comment (TooTallNate) [reply packages/build-utils/src/types.ts:389]: I'm not sure what you mean "twice", but we have `Images` defined in build-utils that we should use instead: https://github.com/vercel/vercel/blob/main/packages/build-utils/src/types.ts#L355-L360

Comment (ijjk) [reply packages/build-utils/src/types.ts:389]: Isn't that the same type/file?

Comment (TooTallNate) [reply packages/build-utils/src/types.ts:389]: Yup. Nvm. I'm blind apparently 🤦

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**